### ISSUE TITLE
problems if IMAGE_BASENAME is changed

### ISSUE THREAD

**Issue by geroschum** [rauc/meta-rauc#64]
When changing the IMAGE_BASENAME in a custom image recipe, I'm facing an issue when defining a RAUC_SLOT of the type image:
The content of `RAUC_SLOT_<slotclass>` is defining the dependency to the image recipe AND it is also used to determine the name of the image file. So if the IMAGE_BASENAME is not matching the recipe name the image file cannot be found.
E.g.
```
RAUC_SLOT_rootfs = "mymachine-myproj-image"
RAUC_SLOT_rootfs[fstype] = "ext4"
```

and inside my recipe (e.g. too shorten the names of the output files)
```
export IMAGE_BASENAME = "myproj-image"
```
rauc is searching for the file
```
mymachine-myproj-image-<machine>.ext4
```
which is not existing.

As a simple solution I patched the bundle.bbclass to handle the slot flag 'file' also for slots of the type image 

```diff --git a/classes/bundle.bbclass b/classes/bundle.bbclass
index 20d5c0b..3c5299a 100644
--- a/classes/bundle.bbclass
+++ b/classes/bundle.bbclass
@@ -143,7 +143,10 @@ def write_manifest(d):
             img_fstype = slotflags.get('fstype')
 
         if imgtype == 'image':
-            imgsource = "%s-%s.%s" % (d.getVar('RAUC_SLOT_%s' % slot), machine, img_fstype)
+            if slotflags and 'file' in slotflags:
+                imgsource = d.getVarFlag('RAUC_SLOT_%s' % slot, 'file')
+            else:
+                imgsource = "%s-%s.%s" % (d.getVar('RAUC_SLOT_%s' % slot), machine, img_fstype)
             imgname = imgsource
         elif imgtype == 'kernel':
             # TODO: Add image type support
```
Now it is possible to define the image file.
```
RAUC_SLOT_rootfs[file] = "myproj-image-${MACHINE}.ext4"
```

What is the correct solution? And why is the flag 'file' not evaluated for 'image'?

**Comment by ejoerns** [comment]
@geroschum Hm... why does this look so familiar to me. Maybe because of https://github.com/ejoerns/meta-rauc/commit/304d3a74969e414ba370dbd7bd4e7a8036ae53a9 :see_no_evil: :/

Thus, yes, I would see it as the 'correct' solution. If you like, please make a PR for it!

**Comment by geroschum** [comment]
Thanks for your answer! I have created a PR.

**Comment by ejoerns** [comment]
Thanks! No fixed by #70.
